src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs

```diff
@@ -5258,7 +5258,13 @@ internal static List<CompletionResult> CompleteVariable(CompletionContext contex
             List<CompletionResult> tempResults = new();
 
             var wordToComplete = context.WordToComplete;
+            string scopePrefix = string.Empty;
             var colon = wordToComplete.IndexOf(':');
+            if (colon >= 0)
+            {
+                scopePrefix = wordToComplete.Remove(colon + 1);
+                wordToComplete = wordToComplete.Substring(colon + 1);
```

Comment:
I see wordToComplete in line 5282 but now we modify the wordToComplete before use in the line. Is this intentional? Have we tests for the 5282 line?

Reply:
`wordToComplete` is modified to remove the scope so in scenarios like: `$script:MyVar<Tab>` it becomes `MyVar`. So we can find variables that start with `MyVar`. My test on line 702 covers this scenario because if we didn't remove the `global` scope from the input text, it wouldn't be able to find any variables at all.